### PR TITLE
Don't include the numpy version in dependencies by default

### PR DESCRIPTION
**Pull request by asmeurer**
It will still be included if CONDA_NPY or --numpy is specified, but by
default, numpy is now treated just as any other package.

### CODE REVIEW

**Comment by asmeurer** [comment]
CC @pelson

**Comment by ccordoba12** [comment]
Sorry if this obvious, but why is this change necessary?

**Comment by mwcraig** [comment]
My memory from the discussion with @pelson @ocefpaf and @scopatz about building out a community channel on anaconda.org is that there are (many?) python packages that depend on numpy but don't specify a version. Right now, if a new numpy is released, all of those packages need to be rebuilt because the numpy version is baked into the file name.

The hope was that a change along these lines would eliminate that need while allowing packages that need a specific numpy version to still require that.

**Comment by asmeurer** [comment]
This is not related to the community channel. The issue is that most packages that depend on numpy do not really need to depend on a specific version. That issue only comes up if they use the C API, and even then, if they use a part of the ABI that has changed.  However, this default requires anyone who wants to support multiple versions of numpy to multiply their build matrix by the number of versions they want to support. Furthermore, whenever a new version of numpy comes out, you have to rebuild all your packages. This can be worked around by using `numpy *` in the recipe, but most people don't realise this.

With this change, it's still possible to do this, by using `--numpy` or `CONDA_NPY`, but the default behavior will be to treat numpy as any other package.

**Comment by pelson** [comment]
I'm totally in support of improving the situation here. Even anaconda shipped packages are getting this wrong (e.g. I believe dask is pure python, and can support np18, yet it claims to be specifically tied to np19 in https://binstar.org/anaconda/dask).

> This can be worked around by using numpy * in the recipe, but most people don't realise this.

So the question is - is it better to improve the documentation, or change the default behaviour?
Given the solution is *so* simple, I can happily live with the former, and I guess the latter does represent a feature change. Is there an equivalently simple way to tell conda to bake in the numpy version without the need to set the CONDA_NPY env var should this PR be merged?

**Comment by asmeurer** [comment]
You can bake the numpy version either by setting CONDA_NPY, using `conda build --numpy`, or setting it explicitly in the recipe (like `numpy 1.9*`).

File: tests/test-recipes/metadata/numpy_build/run_test.bat [hunk 1]
```diff
@@ -0,0 +1,10 @@
+conda list -p "%PREFIX%" --canonical
+if errorlevel 1 exit 1
+for /f "delims=" %%i in ('conda list -p "%PREFIX%" --canonical') do set condalist=%%i
+if errorlevel 1 exit 1
+echo "%condalist%"
+if not "%condalist%"=="conda-build-test-numpy-build-1.0-0" exit 1
+cat "%PREFIX%\conda-meta\conda-build-test-numpy-build-1.0-0.json"
+if errorlevel 1 exit 1
+cat "%PREFIX%\conda-meta\conda-build-test-numpy-build-1.0-0.json" | grep -v 'numpy'
```
Comment (pelson) [inline tests/test-recipes/metadata/numpy_build/run_test.bat]: I wrote a test which did an assertion on in the build json metadata - you might be interested: https://github.com/conda/conda-build/blob/master/tests/test-recipes/metadata/extra_freeform_metadata/run_test.py

Comment (asmeurer) [inline tests/test-recipes/metadata/numpy_build/run_test.bat]: Oh yeah these Windows tests don't work at all. I should probably do something about that.

**Comment by asmeurer** [comment]
OK, I've migrated some of those tests to Python. I also want to try to get Appveyor working so that I can verify that things are working on Windows too.

**Comment by ocefpaf** [comment]
> the default behavior will be to treat numpy as any other package.

:+1: to this!

**Comment by asmeurer** [comment]
Nice, I finally got AppVeyor working! Sorry for mixing in other fixes here and for so many commits (it took a ton of trial and error to make AppVeyor not fail on stupid things).  This now also contains some other miscellaneous fixes for Windows:

- Cygwin git now works correctly.
- The prefix itself is now included in the PATH in the test script on Windows (previously it was just the Scripts directory).

So I am going to merge this.

**Comment by scopatz** [comment]
Glad to see this merged :)

**Pull request merged**